src/components/Menu.tsx

```diff
@@ -0,0 +1,34 @@
+import React, {Dispatch, SetStateAction} from 'react';
+import MenuItem from './MenuItem';
+
+interface MenuProps {
+  toggleMenu: boolean;
+  setToggleMenu: Dispatch<SetStateAction<boolean>>;
+}
+
+const Menu = ({toggleMenu, setToggleMenu}: MenuProps) => {
+  return (
+    <div
+      className={`h-screen w-[30vw] bg-white z-10 fixed ${
+        toggleMenu ? 'top-0 left-0' : 'left-[-30%]'
+      } transition-position duration-500 ease-in-out flex flex-col gap-8 items-center`}
+    >
+      <p className='w-full bg-green-500 h-[4rem] text-white text-xl font-semibold flex justify-between items-center px-6'>
+        Menu
+        <button className='' onClick={() => setToggleMenu(!toggleMenu)}>
```

Comment:
wrap it in a function and pass the function,

`onToggleClick = ( ) => {
...logic
}

onClick={onToggleClick}`